### PR TITLE
RMET-439 File Viewer Plugin ::: As iOS Developer I can open a document

### PR DESCRIPTION
**Pull request by CarlsCorrea**
## Description

Implementetion of the iOS side of the plugin

## Context

Closes: https://outsystemsrd.atlassian.net/browse/RMET-439

## Type of changes

- [ ] Fix (non-breaking change which fixes an issue)
- [x] Feature (non-breaking change which adds functionality)
- [ ] Refactor (cosmetic changes)
- [ ] Breaking change (change that would cause existing functionality to not work as expected)

## Platforms affected
- [ ] Android
- [x] iOS
- [ ] JavaScript

## Checklist

- [x] Pull request title follows the format `RNMT-XXXX <title>`
- [x] Code follows code style of this project
- [x] CHANGELOG.md file is correctly updated
- [x] Changes require an update to the documentation
	- [ ] Documentation has been updated accordingly

### CODE REVIEW

File: plugin.xml [hunk 1]
```diff
@@ -17,11 +17,19 @@
   </platform>
   <platform name="ios">
     <config-file parent="/*" target="config.xml">
-      <feature name="OSFileViewer">
-        <param name="ios-package" value="OSFileViewer"/>
+      <feature name="CDVFileViewer">
```
Comment (vmfo22) [inline plugin.xml:20]: what's the main reason for this name?

Comment (CarlsCorrea) [reply plugin.xml:20]: Gonçalo told me that this is the correct pattern. Start the file that communicates with the Cordova with CVD

**Review by andregrillo — APPROVED** [review]

File: plugin.xml [hunk 2]
```diff
@@ -17,11 +17,19 @@
...
+        <param name="ios-package" value="CDVFileViewer"/>
       </feature>
     </config-file>
-    <source-file src="src/ios/OSFileViewer.m"/>
-    <header-file src="src/ios/OSFileViewer.h"/>
+    <dependency url="https://github.com/agoncalvesos/cordova-plugin-add-swift-support" commit="513e43f94ae94ad609d49f24e7dca23a4c358095" id="cordova-plugin-add-swift-support"/>
```
Comment (vmfo22) [inline plugin.xml:24]: In the future, we might want to have our own version of swift support

File: src/ios/CDVFileViewer.swift [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+        } catch {
+            sendResult(result: "", error: error.localizedDescription)
+        }
+    }
+    
+    @objc(openDocumentFromLocalPath:)
+    func openDocumentFromLocalPath(command: CDVInvokedUrlCommand) {
+        let filePath = command.arguments[0] as? String ?? ""
+        do {
+            try plugin.openDocumentFromLocalPath(url: filePath)
```
Comment (vmfo22) [inline src/ios/CDVFileViewer.swift:41]: If we are passing as a parameter the file path, shouldn't the var name be `filePath` instead of `url`?

Comment (vmfo22) [inline src/ios/CDVFileViewer.swift:41]: Another question, if the filePath is invalid and we get an error to open the document, how are we sending the result back to JS?

Comment (vmfo22) [reply src/ios/CDVFileViewer.swift:41]: For all use cases, an exception will be thrown?

**Review by vmfo22 — CHANGES_REQUESTED** [review]
Nice work @CarlsCorrea! 

Please check my comments

File: src/ios/CDVFileViewer.swift (hunk 3 above)
Comment (CarlsCorrea) [reply src/ios/CDVFileViewer.swift:41]: The error thrown in this case will be "The file does not exist". 
We are mapping empty urls and invalid file path.
If for some reason even the url is not empty and the file exists, and not be possible to open, the error will be "Could not open the document".

Comment (CarlsCorrea) [reply src/ios/CDVFileViewer.swift:41]: sure, make sense I will change the name of the parameter.

File: plugin.xml (hunk 2 above)
Comment (CarlsCorrea) [reply plugin.xml:24]: ok, should I create an issue?

File: src/ios/CDVFileViewer.swift (hunk 3 above)
Comment (vmfo22) [reply src/ios/CDVFileViewer.swift]: Ok, thanks

File: plugin.xml (hunk 2 above)
Comment (vmfo22) [reply plugin.xml:24]: Not in the context of this PR. It's for the future. We need to think about having control of this dependency.

Comment (CarlsCorrea) [reply plugin.xml:24]: ok

File: src/ios/FileViewerOpenDocument.swift [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+    var documentInteractionController:UIDocumentInteractionController?
+    weak var viewController: UIViewController?
+    
+    init(viewController: UIViewController) {
+        self.viewController = viewController
+    }
+
+    func openDocumentFromUrl(url:URL) throws {
+        FileDownloader.downloadfile(url: url, completion: {(success, fileLocationURL) in
+            if success, let filePath = fileLocationURL {
```
Comment (vmfo22) [inline src/ios/FileViewerOpenDocument.swift:21]: If the download fails, how are we dealing with error?

Comment (CarlsCorrea) [reply src/ios/FileViewerOpenDocument.swift:21]: good point, I need to implement the return of the download error...

**Review by vmfo22 — APPROVED** [review]

**Pull request merged**